openedx/core/djangoapps/content_libraries/tasks.py

```diff
@@ -169,6 +174,70 @@ def duplicate_children(
                 self.status.fail({'raw_error_msg': str(exception)})
 
 
+@shared_task(base=LoggedTask)
+@set_code_owner_attribute
+def create_or_update_xblock_upstream_link(usage_key):
+    """
+    Create or update upstream link for a single xblock.
+    """
+    ensure_cms("create_or_update_xblock_upstream_link may only be executed in a CMS context")
+    xblock = modulestore().get_item(UsageKey.from_string(usage_key))
+    if not xblock.upstream or not xblock.upstream_version:
+        TASK_LOGGER.info(f"No upstream or upstream_version found for xblock: {xblock.usage_key}")
```

Comment:
ditto nit: this will generate a lot of log messages when people update their blocks -- I think we should remove this line.
```suggestion
```